### DISCUSSION TITLE
Help with setting up authz endpoints

### DISCUSSION THREAD

**Discussion by sarpuser** [authelia/authelia#7327] [Troubleshooting] [answered]
Hello,

I am wondering if there is a way to access authelia protected services from outside a browser. The best example I can give to this is the LunaSea app with the *arr services. LunaSea is a phone application that you can interact with *arr applications such as radarr or sonarr, using an API key provided by them. Would this just work if I had those services behind authelia? LunaSea also gives you the option to add additional headers to the requests. What would I add here to make this work? Is there even a way to make this work without removing radarr/sonarr from Authelia's umbrella?

Edit: I learned that I need to use autz endpoints for this, but the documentation is kind of confusing. Any help would be appreciated.

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

To access services protected by Authelia from outside a browser, such as through the LunaSea app for interacting with *arr services like Radarr or Sonarr, you would typically need a method to handle authentication programmatically since Authelia is designed primarily for interactive web browser scenarios.

Authelia does not directly support API keys for bypassing authentication in the way *arr services do. However, there are a couple of approaches you could consider:

1. **Dedicated User Account with Simplified Authentication**: You could create a dedicated user account in Authelia for API access and configure it to use a simpler authentication method, such as a single-factor authentication (1FA), from specific IPs that you control. This method was suggested in a discussion on Authelia's GitHub page for accessing protected endpoints with automated scripts. You would then use the basic authentication scheme with the Proxy-Authorization header for your requests from LunaSea.

2. **OAuth2 Tokens (Future Implementation)**: Authelia is planning to introduce an OAuth2 mechanism to handle scenarios like this (see GitHub discussion #5916). Once implemented, you could use OAuth2 tokens to authenticate API requests, which would be a more secure and scalable solution than using basic authentication or IP-based restrictions.

For now, if you decide to use the first approach, you would need to configure your reverse proxy (such as Traefik or HAProxy) to handle these authentication headers and possibly set up Authelia to accept simpler authentication from the IPs associated with your LunaSea app.

In LunaSea, if you have the option to add additional headers, you would add the Proxy-Authorization header with the credentials of the dedicated user account you set up for API access. Make sure this user's credentials are securely handled and that the connection to your services is encrypted (using HTTPS).

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment, marked as answer]
Generally I'd recommend using the integration setup detailed for your proxy here: https://www.authelia.com/integration/proxies/support/

Then from memory if you add a bypass rule for the *arr apps it works correctly. As those endpoints are secured by the API key they are still secure with that particular setup. Example here: https://www.authelia.com/configuration/security/access-control/#resources

**Reply by james-d-elliott** [reply]
Also sorry for not replying sooner.
